src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
             _edmType = new EdmDeltaCollectionType(new EdmEntityTypeReference(_entityType, isNullable: true));
             _edmTypeReference = new EdmCollectionTypeReference(_edmType);
         }
+
+        /// <summary>
+        /// Patch for Types without underlying CLR types
+        /// </summary>
+        /// <param name="originalCollection"></param>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(ICollection<EdmStructuredObject> originalCollection)
+        {            
+            PatchHandler = new DefaultTypelessPatchHandler(originalCollection, _entityType);
+
+            return CopyChangedValues();
+        }
+
+        /// <summary>
+        /// Handler for users Create, Get and Delete Methods
+        /// </summary>
+        internal TypelessPatchMethodHandler PatchHandler { get; set; }
+
+        /// <summary>
+        /// Patch for EdmChangedObjectCollection, a collection for IEdmChangedObject 
+        /// </summary>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(TypelessPatchMethodHandler patchHandler)
+        {
+            PatchHandler = patchHandler;
+            return CopyChangedValues();
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+        internal EdmChangedObjectCollection CopyChangedValues()
+        {
+            EdmChangedObjectCollection changedObjectCollection = new EdmChangedObjectCollection(_entityType);
+            List<IEdmStructuralProperty> keys = _entityType.Key().ToList();
+
+            foreach (IEdmChangedObject changedObj in Items)
```

Comment:
This foreach block looks very much like another in _src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs_ [here](https://github.com/OData/WebApi/pull/2455/files#diff-f4621f7e9e09c9dca539d23c90a367ffc0021789f817b128d4c1787f0ab51b9aR93). Can any of the logic be extracted into a method such that it can be shared by both methods?

Reply:
Very unlikely as one is edm (typeless) and other is based on Type <TStructuralType>